xtc/src/history.rs

```diff
@@ -71,12 +75,72 @@ impl HistoryBuffer {
     pub fn len(&self) -> u64 {
         self.history.len()
     }
+
+    pub fn get_history(&self) -> &History {
+        &self.history
+    }
 }
 
+//////////////////// BEGIN OF ERC-20 ///////////////////////
+
+#[update]
+pub async fn getTransaction(index: Nat) -> TxRecord {
```

Comment:
Feel free to change the function name, I had to make it different than the other `get_transaction` and I don't have as good of an understanding of this project's internals.

```suggestion
#[update(name = "getTransaction")]
pub async fn get_transaction_erc(index: Nat) -> TxRecord {
```